✨(social) improve social sharing with og:description

igobranco opened the pull request:
Improve social sharing by specifying og:description meta for blogpost, category, person and program.

* if blog defaults to blog excerpt placeholder
* if category defaults to description placeholder
* if person defaults to bio placeholder
* if program defaults to excerpt placeholder
* if organization defaults to organization description placeholder value

sampaccoud commented on CHANGELOG.md:
```diff
@@ -17,6 +17,8 @@ Versioning](https://semver.org/spec/v2.0.0.html).
 - Add a `outline` variant to button
 - Create a `shadowed-box` mixin
+- Improve social sharing by specifying og:description meta
+  for blogpost, category, person and program.
```
+ organization

igobranco replied:
done

sampaccoud commented on src/richie/apps/courses/templates/courses/cms/blogpost_detail.html:
```diff
@@ -13,6 +13,9 @@
         <meta property="og:image:height" content="{{ thumb.height }}" />
     {% endblockplugin %}
+    {% if not current_page|is_empty_placeholder:"excerpt" %}
+        <meta property="og:description" content="{% filter striptags %}{% show_placeholder 'excerpt' current_page %}{% endfilter %}" />
```
I think it should be trimed and trunked the same as the meta description.

igobranco replied:
Yes. Similar approach. Trunked to 200 characters.

sampaccoud commented:
@igobranco If I'm not mistaken, you forgot to test stripping tags and trimming like you did on meta description?

igobranco commented:
> @igobranco If I'm not mistaken, you forgot to test stripping tags and trimming like you did on meta description?

I think I didn't forget nothing.
1. Test a normal string on the og:description
2. Test a max length of the meta og:description
3. Test the absent of the og:description if placeholder not exists

Depending on the placeholder type, the template implementation differs:
1. if it's a plain text, I'm not trimming and not stripping tags (because they don't exist). 
2. if it uses the `CKEditorPlugin` then we are stripping tags and trimming.

The `CKEditorPlugin` adds a new line at the end, so when we test a placeholder that use it, we are testing that a new line is not being added to the end. Nevertheless, I added spaces to the normal test, to test that they are being trimmed. From: `body=" A further <b>description</b> of the category  ",`. to `'<meta property="og:description" content="A further description of the category" />'`

I could also include the `trim` filter on all the type of pages (that use the plain text placeholder).

---
I've also detected that the organization description placeholder should use `CKEditorPlugin` instead of `PlainTextPlugin`. So I fixed the meta description tests.

sampaccoud commented:
> I've also detected that the organization description placeholder should use `CKEditorPlugin` instead of `PlainTextPlugin`. So I fixed the meta description tests.

Ah! Yes, that's the one I was looking at. 😉

sampaccoud commented on tests/apps/courses/test_templates_organization_detail.py:
```diff
@@ -24,6 +24,84 @@ class OrganizationCMSTestCase(CMSTestCase):
+            language="en",
+            placeholder=placeholder,
+            plugin_type="CKEditorPlugin",
+            body="A longe description of the organization  ",
```
This is where you should have html text no?

igobranco replied:
Yes, sorry. I've missed that.

sampaccoud reviewed (approved):
Gooo!
